octue/cloud/pub_sub/service.py

```diff
@@ -50,10 +57,11 @@ class Service:
     :param str|None service_id: a unique ID to give to the service (any string); a UUID is generated if none is given
     :param callable|None run_function: the function the service should run when it is called
     :param str|None name: an optional name to use for the service to override its ID in its string representation
+    :param iter(dict)|None service_registries: the names and endpoints of the registries used to resolve service revisions when asking questions; these should be in priority order (highest priority first)
     :return None:
     """
 
-    def __init__(self, backend, service_id=None, run_function=None, name=None):
+    def __init__(self, backend, service_id=None, run_function=None, name=None, service_registries=None):
```

Comment:
I believe that as we move forward, the `backend` here should be able to be initialised from the service registry response.

Reply:
Yep I agree. Shall I leave any implementation of that until registries support that?

Reply:
Yes, absolutely leave that for now.